app/experimenter/nimbus-ui/src/services/hooks.ts

```diff
@@ -0,0 +1,48 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { useEffect, useState } from "react";
+
+/**
+ * Hook to trigger a browser dialog when a user attempts to exit the page.
+ *
+ * Example:
+ *
+ * const shouldWarnOnExit = useExitWarning();
```

Comment:
What's the use case for `shouldWarnOnExit(false)`, wouldn't we just not use this hook? If we set `useExitWarning` to default to true, we don't even need `const shouldWarnOnExit = useExitWarning();`, we can simply use 

```
          <button
            onClick={() => {
              useExitWarning();
            }}
          >
```

Reply:
Good question. We need to keep a reference to the hook so when the user clicks the "Next" button (or whatever the save and continue button is) we can turn off this exit warning.

Reply:
Since we'll have a "Save" button and a "Next" button which is essentially "Save and navigate," I think we'll only need this in the navigation right? We could have something like, a `hasMadeChanges` var that sets to `true` onChange, and if that's true, use the hook on navigation attempt. At least, this is what I thought "Next" would be. Maybe it doesn't also "Save"?

Reply:
Yeah, that def could work. Isn't that more or less the same level of effort than setting the hook up at the top and setting it to true or false?

Reply:
I'm going to go ahead and r+ since this shouldn't block the PR merge, if we want to simplify this later when we actually implement it we can.